src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,41 @@
 
 class Animal {
   // write your code here
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+    Animal.alive.push(this);
+  }
+
+  static alive = [];
 }
 
 class Herbivore extends Animal {
   // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
   // write your code here
+
+  bite(prey) {
+    if (prey.hidden === false && prey instanceof Herbivore) {
+      prey.health -= 50;
+    }
+
+    if (prey.health <= 0) {
+      Animal.alive.splice(
+        Animal.alive.indexOf(
+          animal => Animal.alive.animal.name === prey.name), 1);
```

Comment:
checklist point 2 - Try to avoid nesting loops (indexOf + splice), use filter instead